### PR TITLE
BRMS 281: Adding/update error response to include response body with code and description

### PR DESCRIPTION
**Pull request by chris-ianson**
@manishsw @adamconder 

Updated error responses with relevant messages. We've had to update HeaderValidator so we can respond with specific error messages.

We still need to update the api documentation but can be reviewed

### CODE REVIEW

File: app/uk/gov/hmrc/brm/models/brm/HttpResponse.scala [hunk 1]
```diff
@@ -57,78 +58,78 @@ trait HttpResponse {
 
 object CustomErrorResponse extends HttpResponse {
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/models/brm/HttpResponse.scala]: @chris-ianson We can remove this duplicate object and just call ErrorResponses.handle directly.

File: app/uk/gov/hmrc/brm/models/brm/HttpResponse.scala [hunk 2]
```diff
@@ -57,78 +58,78 @@ trait HttpResponse {
...
-                              ) extends HttpResponseBody
\ No newline at end of file
+object ErrorResponses {
+
+  type ErrorResponses = List[(String, HttpResponseBody)]
+
+  def handle(key: String, error: String) = {
+    error match {
+      case "error.path.missing" =>
+        BadRequest("")
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/models/brm/HttpResponse.scala]: @chris-ianson Should we implement a generic Bad_Request response body?

File: test/uk/gov/hmrc/brm/models/ErrorResponseSpec.scala [hunk 3]
```diff
@@ -30,53 +32,15 @@ class ErrorResponseSpec extends UnitSpec with WithFakeApplication {
...
-
-      val code = 123456789
-      val response = ErrorResponse.getErrorResponseByErrorCode(code)
+    "return BadRequest with empty body when key doesn't exist" in {
 
-      response shouldBe a[JsObject]
-      (response \ "details").as[JsString].value should be("something is wrong")
+      val response = await(ErrorResponses.handle("firstNameInvalid", ""))
+      bodyOf(response) shouldBe empty
+      response.header.status shouldBe BAD_REQUEST
```
Comment (adamconder) [inline test/uk/gov/hmrc/brm/models/ErrorResponseSpec.scala]: @chris-ianson Can we add a couple of unit tests just for the other ones? Maybe 1 or two?

File: test/uk/gov/hmrc/brm/utils/HeaderValidatorSpec.scala [hunk 4]
```diff
@@ -52,18 +52,19 @@ class HeaderValidatorSpec extends UnitSpec with OneAppPerSuite with MockitoSugar
...
+//      HeaderValidator.versionValidation() shouldBe false
+//      HeaderValidator.contentTypeValidation() shouldBe false
+//      HeaderValidator.auditSourceValidation() shouldBe false
+//    }
+
+//    "return false when Accept header is invalid" in {
+//      HeaderValidator.acceptHeaderValidation(accept = Some("application/vNd.HMRC.1.0+xml")) shouldBe false
+//    }
+//
+//    "return false when Audit Source is invalid" in {
```
Comment (adamconder) [inline test/uk/gov/hmrc/brm/utils/HeaderValidatorSpec.scala]: @chris-ianson remove commented code.

File: test/uk/gov/hmrc/brm/models/ErrorResponseSpec.scala [hunk 5]
```diff
@@ -20,29 +20,45 @@ import akka.stream.Materializer
...
       response.header.status shouldBe BAD_REQUEST
     }
 
+//    "return BadRequest with generic body when firstName key is missing" in {
+//      val response = await(ErrorResponses.getHttpResponse("firstName", "error.path.missing"))
+//      bodyOf(response) shouldBe MockErrorResponses.BAD_REQUEST.json
+//      response.header.status shouldBe BAD_REQUEST
+//    }
+//
+//    "return BadRequest with specific body when lastName is invalid" in {
```
Comment (adamconder) [inline test/uk/gov/hmrc/brm/models/ErrorResponseSpec.scala]: @chris-ianson remove commented code.

**Review by adamconder — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -123,6 +123,90 @@ Content-Type: application/json; charset=utf-8
...
+
+```http
+HTTP/1.1 406 Not Acceptable
+Accept: application/vnd.hmrc.1.0+xml
+Cache-Control: no-cache,no-store,max-age=0
+Content-Type: application/json; charset=utf-8
+
+{
+  "code": "INVALID_CONTENT_TYPE",
+  "message": "Accept header is invalid.""
```
Comment (adamconder) [inline README.md]: @chris-ianson Duplicate "" on the end here :D

File: README.md [hunk 7]
```diff
@@ -123,6 +123,90 @@ Content-Type: application/json; charset=utf-8
...
+
+```http
+HTTP/1.1 406 Not Acceptable
+Accept: application/vnd.hmrc.12.0+xml
+Cache-Control: no-cache,no-store,max-age=0
+Content-Type: application/json; charset=utf-8
+
+{
+  "code": "INVALID_CONTENT_TYPE",
+  "message": "Accept header is invalid.""
```
Comment (adamconder) [inline README.md]: Duplicate string.

File: README.md [hunk 8]
```diff
@@ -123,6 +123,90 @@ Content-Type: application/json; charset=utf-8
 }
 ```
```
Comment (adamconder) [inline README.md:125]: @chris-ianson We need to add the body example for the generic BadRequest

**Review by adamconder — CHANGES_REQUESTED** [review]

File: README.md (hunk 8 above)
Comment (chris-ianson) [reply README.md:125]: @manishsw updated and pushed

**Pull request merged**